### DISCUSSION TITLE
How to have code output (a figure) on a separate slide

### DISCUSSION THREAD

**Discussion by jlperla** [quarto-dev/quarto-cli#6918] [Q&A]
### Description

I am generating a set of beamer slides - with an accompanying jupyter notebook with a fair amount of code examples of plotting. Right now my pattern is the I am doing something like the following:
    
    ## Plotting $k_t$ vs. $k_{t+1}$
    ```{python}
    #| eval: false
    k_grid = np.linspace(0, 1.5, 100)
    fig, ax = plt.subplots()
    ax.plot(k_grid, g(k_grid), label="$g(k_t)$")
    ax.plot(k_grid, k_grid, label="$k_{t+1}=k_t$")
    ax.set_xlabel("$k_t$")
    ax.set_ylabel("$k_{t+1}$")
    ax.legend()
    plt.show()
    ```
    
    ## Plotting $k_t$ vs. $k_{t+1}$
    ```{python}
    #| echo: false
    k_grid = np.linspace(0, 1.5, 100)
    fig, ax = plt.subplots()
    ax.plot(k_grid, g(k_grid), label="$g(k_t)$")
    ax.plot(k_grid, k_grid, label="$k_{t+1}=k_t$")
    ax.set_xlabel("$k_t$")
    ax.set_ylabel("$k_{t+1}$")
    ax.legend()
    plt.show()
    ```

Is there a better way to do this that copy/pasting all of the code into two separate blocks?  It is fine to me if the output always has the slide name from the previous section.... or even had no title as a worst case.

**Comment by mcanouil** [comment]
See <https://quarto.org/docs/presentations/revealjs/#output-location>/

**Comment by jlperla** [comment]
Thanks.  Alas, this is for beamer.  Is there an equivalent there?  I tried just putting
    
    ## Plotting $k_t$ vs. $k_{t+1}$
    ```{python}
    #| output-location: slide
    k_grid = np.linspace(0, 1.5, 100)
    fig, ax = plt.subplots()
    ax.plot(k_grid, g(k_grid), label="$g(k_t)$")
    ax.plot(k_grid, k_grid, label="$k_{t+1}=k_t$")
    ax.set_xlabel("$k_t$")
    ax.set_ylabel("$k_{t+1}$")
    ax.legend()
    plt.show()
    ```
And no luck

**Reply by mcanouil** [reply]
~~I had no way to know since a complete and reproducible document was not provided 🤷‍♂️~~
My bad, it was written beamer in the text. Note that sharing a complete document is always more useful than a small snippet of code.

Via the knitr engine, there is the concept of code cell reuse (<https://bookdown.org/yihui/rmarkdown-cookbook/reuse-chunks.html>) but I don't think it even exist in Jupyter/Python.

**Comment by jlperla** [comment]
Sorry about that.  The full code which compiles.  My current workaround:
    
    ---
    title: "Test Doc"
    format: beamer 
    execute: 
      cache: true    
    echo: true
    ---
    
    ## Some Code
    
    Here is some python code
    
    ```{python}
    #| eval: false
    import matplotlib.pyplot as plt
    
    plt.plot([1, 2, 3, 4])
    plt.ylabel('some numbers')
    plt.show()
    ```
    
    ## Results
    
    ```{python}
    #| echo: false
    import matplotlib.pyplot as plt
    
    plt.plot([1, 2, 3, 4])
    plt.ylabel('some numbers')
    plt.show()
    ```

The revealjs with `output-location:slide` works great:

    ---
    title: "Test Doc"
    format: revealjs 
    execute: 
      cache: true    
    echo: true
    ---
    
    ## Some Code
    
    Here is some python code
    
    ```{python}
    #| output-location: slide
    import matplotlib.pyplot as plt
    
    plt.plot([1, 2, 3, 4])
    plt.ylabel('some numbers')
    plt.show()
    ```
BUt it is revealjs specific.  Ignored if you swap that to beamer

I tried various permutaitons on the `chunk` stuff, but it didn't help.  Also, even things like

    ```{python, eval=FALSE}
    import matplotlib.pyplot as plt
    
    plt.plot([1, 2, 3, 4])
    plt.ylabel('some numbers')
    plt.show()
    ```

Don't work.  You need the manual `#| eval: false`


Not sure if there are any other possible hacks I am missing?

**Reply by mcanouil** [reply]
Don't think so or maybe there are "cell magic" in Python/Jupyter. (I am no expert in Jupyter)

**Reply by cscheid** [reply]
> Don't work. You need the manual #| eval: false

This is not a "manual" declaration; that's the proper way to declare these in Quarto. (The knitr style is supported as legacy syntax, but it's not a Quarto feature)
